docs/BINDING_PROCESS_DIAGRAM.md

```diff
@@ -0,0 +1,148 @@
+│                     │ <──────────────────────────────── │  SuppIsReadyToSendConfirm   │
+│                     │                                   │  - Received Accept          │
+│                     │                                   │  - Ready to send Confirm    │
+└─────────────────────┘                                   └─────────────┬───────────────┘
+                                                                          │
+                                                                          │ cast_confirm_accept()
+                                                                          ▼
+                                                              ┌─────────────────────────────┐
+                                                              │  State Decision Point       │
+                                                              │  - If addenda: SendAddenda  │
+                                                              │  - If no addenda: Complete  │
+                                                              └─────────────┬───────────────┘
+                                                                            │
+                                                                            │ Yes: cast_addenda() / No: Complete
+                                                                            ▼
+┌─────────────────────┐    Binding Complete               ┌─────────────────────────────┐
+│  RespHasBoundAs-    │ <───────────────────────────────  │  SuppHasBoundAsSupplicant   │
+│   Respondent        │                                  │  - Binding Complete         │
+│                     │                                  │  - Nothing more to do       │
+└─────────────────────┘                                  └─────────────────────────────┘
+```
+
+## Method Flow by Role
+
+### Supplicant Flow (Device initiating binding)
+```python
+async def initiate_binding_process(self, offer_codes, confirm_code=None, ratify_cmd=None):
+    1. _make_offer(offer_codes, oem_code)           # Send Offer packet
+    2. _wait_for_accept(tender)                     # Wait for Accept response
+    3. _confirm_accept(accept, confirm_code)        # Send Confirm packet
+    4. _cast_addenda(accept, ratify_cmd)            # Optional: Send Addenda
+
+    # Result: SuppHasBoundAsSupplicant state
+```
+
+### Respondent Flow (Device responding to binding)
+```python
+async def wait_for_binding_request(self, accept_codes, idx="00", require_ratify=False):
+    1. _wait_for_offer()                            # Wait for Offer packet
+    2. _accept_offer(tender, accept_codes, idx)     # Send Accept packet
+    3. _wait_for_confirm(accept)                    # Wait for Confirm packet
+    4. _wait_for_addenda(accept)                    # Optional: Wait for Addenda
+
+    # Result: RespHasBoundAsRespondent state
+```
+
+## State Details
+
+| State | Role | Description | Key Methods |
```

Comment:
Also nice. I could not get this type of table to display in sphinx, see ramses_rf/database.py doc. Fine for now